generated/nixnetsocket/nixnetsocket.proto

```diff
@@ -186,7 +195,7 @@ message AddrInfo {
 }
 
 message AddrInfoHint {
-  GetAddrInfoFlags flags = 1;
+  int32 flags = 1;
```

Comment:
This field and flags on `AddrInfo` either needed to be an `int32` or `repeated GetAddrInfoFlags` to allow for a combination of flags. I'm honestly not sure which is better but I went with int32 since it was simpler and allows the client to do bitwise operations on the flags / construct them how they would for the C API.

If, however, the client just wants to add/remove flags and leave the bitwise combination to us we could use repeated Flags enum and do the translations to and from an int in the conversion code.

Reply:
I created AB#1896770 that's related to this. I'm leaning towards exposing both and |-ing it all together. See discussion in #626 ( 'note: FrameFlags is used as "bitfield_as_enum_array" here. ').

Note that the codgen for `bitfield_as_enum_array` is only for outputs. That PR is adding a custom code variant "inspired by" bitfield_as_enum_array. We might be somewhere in between (or have some of each) with the different bitfields on nxsocket.

Reply:
That sounds good. I did that for AddrInfoHint and AddrInfo, so now they have both `repeated GetAddrInfoFlags flags = 1` and `int32 flags_raw = 2` fields. Conversion code was updated to | the fields for input (AddrInfoHint). For output, I set `flags_raw` to equal the flags gotten from the driver API and then added a helper `convert_to_addr_info_flags` to take the int flags and fill out the repeated `flags` field based on it. Updated tests to test both `flags` and `flags_raw` and added new test to directly test the new helper.